core/modules/simpletest/tests/icon.test

```diff
@@ -90,6 +90,37 @@ class IconTestCase extends BackdropWebTestCase {
     $this->assertIconsEqual($core_path, 'gear-bold', FALSE, 'Icon provided by a base-theme uses the correct icon.');
   }
 
+  /**
+   * Test that rendered icons have the expected content and attributes.
+   */
+  protected function testIconAttributes() {
+    $options_alt = array(
+      'alt' => 'Hello world',
+    );
+    $result_alt = '<svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 256 256" fill="currentColor" class="icon icon--circle-fill"><title>Hello world</title><path d="M232,128A104,104,0,1,1,128,24,104.13,104.13,0,0,1,232,128Z"/></svg>';
+    $icon = icon('circle-fill', $options_alt);
+    $this->assertEqual($icon, $result_alt, 'Alt option properly converted to title dom element.');
```

Comment:
Can we please change all the `$result_*` variables to `$expected_*` instead? Makes it easier to understand. If anything, I would expect `$result_*` to be the output of `icon()`; so this gets confusing for a few seconds while reading it.

Other than that, these tests look great 👍🏼 ...can we please add another one that handles the case where the svg already has a `<title>` and what happens when we try to add `alt` to it? I know that the core-provided Phosphor icons don't have a `<title>`, but `image_add_svg_attributes()` is meant to be generic and used with other icons and icon sets that might have one (or Phosphor might change their icons to have one in the future).

Reply:
...this is what I'd expect:
```suggestion
    $expected_alt = '<svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 256 256" fill="currentColor" class="icon icon--circle-fill"><title>Hello world</title><path d="M232,128A104,104,0,1,1,128,24,104.13,104.13,0,0,1,232,128Z"/></svg>';
    $result = icon('circle-fill', $options_alt);
    $this->assertEqual($result, $expected_alt, 'Alt option properly converted to title DOM element.');
```